test/Google.Datastore.V1Beta3.Tests/EntityTest.cs

```diff
@@ -77,5 +77,32 @@ public void Mutations_NoKey()
             Assert.Throws<InvalidOperationException>(() => entity.ToUpsert());
             Assert.Throws<InvalidOperationException>(() => entity.ToDelete());
         }
+
+        [Fact]
+        public void ArrayProperties()
+        {
+            // Demonstrating three ways of specifying arrays as entity properties
+            var variant1 = new Entity
+            {
+                ["tags"] = new[] { "a", "b", "c" }
```

Comment:
This looks very nice.

Can you elaborate regarding your concerns about null values?